src/main/java/com/example/doodle_clone/models/user/CustomUserDetailsService.java

```diff
@@ -0,0 +1,22 @@
+package com.example.doodle_clone.models.user;
+
+import com.example.doodle_clone.repo.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+
+public class CustomUserDetailsService implements UserDetailsService {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userRepository.findByEmailId(username);
+        if(user==null){
```

Comment:
wrap with optional instead of null check

Reply:
fix